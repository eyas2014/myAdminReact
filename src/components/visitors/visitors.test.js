import React from 'react';
import Visitors from './visitors';
import enzyme from 'enzyme';

jest.mock('react-transition-group');
jest.mock('./world-map.jsx');
jest.mock('./sparkline.jsx');
const data={worldMap: 'worldmap', sparkLine: 'sparkline'};

document.body.innerHTML='<div  id="root"></div>';

const wrapper=enzyme.mount(<Visitors data={data}></Visitors>, {attachTo: document.getElementById('root')});

it ('should renders correct htmls', ()=>{
	expect(document.getElementById('root').innerHTML).toMatchSnapshot();
	expect(document.getElementsByClassName('wrapper')[0].innerHTML).toEqual('<main></main><footer></footer>');
});






it ('CSSTransition should be passed with correct args', ()=>{
	let transitionParam={ in: true,
    						classNames: 'componentCollapse',
    						timeout: 500,
      						unmountOnExit: true };
	expect(require('react-transition-group').calledWith()).toEqual(transitionParam);

	let toggleButton=wrapper.find('header div button');
	toggleButton.at(1).simulate('click');
	transitionParam.in=false;
	expect(require('react-transition-group').calledWith()).toEqual(transitionParam);
});



it('wrapper should collapse after clicking toggle button', ()=>{
	console.log(document.getElementById('root').innerHTML);


});


