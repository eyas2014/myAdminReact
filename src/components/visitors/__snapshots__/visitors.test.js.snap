// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correct htmls 1`] = `"<div class=\\"visitors\\"><header class=\\"ui-sortable-handle\\"><div class=\\"btn\\"><i class=\\"fa fa-map-marker\\"></i><span> Visitors</span></div><div class=\\"float-right tools\\"><button class=\\"btn btn-sm\\"><i class=\\"fa fa-calendar\\"></i></button><button class=\\"btn btn-sm ml-1\\"><i class=\\"fa fa-minus\\"></i></button></div></header><div class=\\"wrapper\\"><main></main><footer></footer></div></div>"`;
